lib/widgets/subscription_list.dart

```diff
@@ -188,8 +188,12 @@ class _SubscriptionList extends StatelessWidget {
       itemBuilder: (BuildContext context, int index) {
         final subscription = subscriptions[index];
         final unreadCount = unreadsModel!.countInChannel(subscription.streamId);
-        // TODO(#712): if stream muted, show a dot for unreads
-        return SubscriptionItem(subscription: subscription, unreadCount: unreadCount);
+        final hasUnmutedUnreads = unreadCount > 0;
+        final showMutedUnreadBadge = !hasUnmutedUnreads && unreadsModel!.hasMutedInChannel(subscription.streamId);
```

Comment:
As I suggested at https://github.com/zulip/zulip-flutter/pull/714#discussion_r1694343677, I think this logic will be cleaner if it uses the existing method `countInChannelNarrow` (previously `countInStreamNarrow`).